ceph-releases/luminous/ubuntu/16.04/daemon/zap_device.sh

```diff
@@ -1,8 +1,102 @@
 #!/bin/bash
 set -e
 
+# Get a list of child devices from a root device
+function get_child_partitions {
+# $1: raw parent device
+  if [ ! -e  "${1}" ] && [ ! -b "${1}" ]; then
+    log "Error, ${1} doesn't seem to be a valid device!"
+    exit 1
+  fi
+  local parts
+  parts=$(lsblk -no KNAME "${1}")
+  for p in $parts; do
+    kname=$(lsblk --nodeps -no KNAME /dev/"${p}")
+    pkname=$(lsblk --nodeps -no PKNAME /dev/"${p}")
+    if [ "${kname}" != "${pkname}" ] && [ -n "${pkname}" ]; then
+      echo /dev/"${p}"
+    fi
+  done
+}
+
+function get_dmcrypt_uuid_part {
+  # look for Ceph encrypted partitions
+  # Get all dmcrypt for ${device}
+  blkid -t TYPE="crypto_LUKS" "${1}"* -o value -s PARTUUID
+}
+
+function get_opened_dmcrypt {
+  # Get actual opened dmcrypt for ${device}
+  dmsetup ls --exec 'basename' --target crypt
+}
+
+function zap_dmcrypt_device {
+  # $1: list of cryptoluks partitions (returned by get_dmcrypt_uuid_part)
+  # $2: list of opened dm (returned by get_opened_dmcrypt)
+  if [ $# -ne 2 ]; then
```

Comment:
I don't understand the use case for this. YOU are calling this function so you are passing the args, aren't we going to always have 2 args?